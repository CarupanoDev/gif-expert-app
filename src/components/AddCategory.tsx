import { ChangeEvent, Dispatch, FormEvent, SetStateAction, useState } from "react"

export const AddCategory = (
  { setCategories }:
    {
      setCategories: Dispatch<SetStateAction<string[]>>
    }) => {

  const [inputValue, setInputValue] = useState<string>("")

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.target.value)
  }

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategories((categories: any) => [inputValue, ...categories])
      setInputValue("")
    }
  }

  return (

    <form onSubmit={handleSubmit} action="">
      <p>{inputValue}</p>
      <input
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  )
}
