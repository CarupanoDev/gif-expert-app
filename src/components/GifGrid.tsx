import { GifGridItem } from "./GifGridItem"

import { useFetchGifs } from "../hooks/useFetchGifs"

interface GifGridProp {
  category: string
}

export const GifGrid = ({ category }: GifGridProp ) => {

  const { data: images, loading } = useFetchGifs( category )

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}

      <div className="card-grid">
          {
            images.map( image => (
              <GifGridItem
                // @ts-ignore
                key = {image.id}
                {...image}
              />
            ))
          }
      </div>
    </>
  )
}
