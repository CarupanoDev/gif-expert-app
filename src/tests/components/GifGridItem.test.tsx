import { describe, expect, test } from "vitest"
import { shallow } from "enzyme"
import { GifGridItem } from "../../components/GifGridItem"

describe("Test on <GifGridItem/>", () => {
  const title = "The best app"
  const url = "https://localhost/algo.jpg"
  const wrapper = shallow(<GifGridItem title={title} url={url} />)

  test("It should show the component correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  test("It should have a paragraph with the title", () => {
    const p = wrapper.find("p")
    expect(p.text().trim()).toBe(title)
  })

  test("It should have the same img and url from props", () => {

    const img = wrapper.find("img")

    expect(img.props().src).toBe(url)
    expect(img.props().alt).toBe(title)

  })

  test('Must have animate__fadeIn', () => {
    const div = wrapper.find('div');

    expect(div.hasClass('animate__fadeIn')).toEqual(true);

  })
})
