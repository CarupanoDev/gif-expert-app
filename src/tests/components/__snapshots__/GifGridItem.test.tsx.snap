// Vitest Snapshot v1

exports[`Test on <GifGridItem/> > It should show the component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="The best app"
    src="https://localhost/algo.jpg"
  />
  <p>
    The best app
  </p>
</div>
`;
